src/cmd/services/m3comparator/main/querier.go

```diff
@@ -0,0 +1,218 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/m3db/m3/src/dbnode/encoding"
+	"github.com/m3db/m3/src/dbnode/ts"
+	"github.com/m3db/m3/src/query/block"
+	"github.com/m3db/m3/src/query/storage"
+	"github.com/m3db/m3/src/query/storage/m3"
+)
+
+var _ m3.Querier = (*querier)(nil)
+
+type querier struct {
+	encoderPool   encoding.EncoderPool
+	iteratorPools encoding.IteratorPools
+	sync.Mutex
+}
+
+func noop() error { return nil }
+
+type seriesBlock []ts.Datapoint
+type tagMap map[string]string
+type series struct {
+	blocks []seriesBlock
+	tags   tagMap
+}
+
+func generateSeriesBlock(
+	start time.Time,
+	blockSize time.Duration,
+	resolution time.Duration,
+) seriesBlock {
+	numPoints := int(blockSize / resolution)
+	dps := make(seriesBlock, 0, numPoints)
+	for i := 0; i < numPoints; i++ {
+		dp := ts.Datapoint{
+			Timestamp: start.Add(resolution * time.Duration(i)),
+			Value:     rand.Float64(),
+		}
+
+		dps = append(dps, dp)
+	}
+
+	return dps
+}
+
+func generateSeries(
+	start time.Time,
+	end time.Time,
+	blockSize time.Duration,
+	resolution time.Duration,
+	tags tagMap,
+) (series, error) {
+	numBlocks := int(math.Ceil(float64(end.Sub(start)) / float64(blockSize)))
+	if numBlocks == 0 {
+		return series{}, fmt.Errorf("comparator querier: no blocks generated")
+	}
+
+	blocks := make([]seriesBlock, 0, numBlocks)
+	for i := 0; i < numBlocks; i++ {
+		blocks = append(blocks, generateSeriesBlock(start, blockSize, resolution))
+		start = start.Add(blockSize)
+	}
+
+	return series{
+		blocks: blocks,
+		tags:   tags,
+	}, nil
+}
+
+func (q *querier) generateOptions(
+	start time.Time,
+	blockSize time.Duration,
+) iteratorOptions {
+	return iteratorOptions{
+		start:         start,
+		blockSize:     blockSize,
+		encoderPool:   q.encoderPool,
+		iteratorPools: q.iteratorPools,
+	}
+}
+
+type seriesGen struct {
+	name string
+	res  time.Duration
+}
+
+// FetchCompressed fetches timeseries data based on a query.
+func (q *querier) FetchCompressed(
+	ctx context.Context,
+	query *storage.FetchQuery,
+	options *storage.FetchOptions,
+) (m3.SeriesFetchResult, m3.Cleanup, error) {
+	var (
+		// TODO: take from config.
+		blockSize = time.Hour * 12
+		start     = query.Start.Truncate(blockSize)
+		end       = query.End.Truncate(blockSize).Add(blockSize)
+		opts      = q.generateOptions(start, blockSize)
+
+		// TODO: take from config.
+		gens = []seriesGen{
+			seriesGen{"foo", time.Second},
+			seriesGen{"bar", time.Second * 15},
+			seriesGen{"quail", time.Minute},
+		}
+
+		actualGens []seriesGen
+	)
+
+	q.Lock()
+	rand.Seed(start.Unix())
+	for _, matcher := range query.TagMatchers {
+		// filter if name, otherwise return all.
+		if bytes.Equal([]byte("__name__"), matcher.Name) {
```

Comment:
Maybe use the `model.NameLabel` or w/e the shared const is here?

Reply:
Pulling out of default tag options now,